bundle/config/mutator/translate_paths_jobs.go

```diff
@@ -38,28 +39,48 @@ func (t *translateContext) applyJobTranslations(v dyn.Value) (dyn.Value, error)
 			return dyn.InvalidValue, fmt.Errorf("unable to determine directory for job %s: %w", key, err)
 		}
 
-		rewritePatternFn, err := t.getRewritePatternFn(kind)
+		mode, err := getJobTranslateMode(kind)
 		if err != nil {
 			return dyn.InvalidValue, err
 		}
 
-		return t.rewriteRelativeTo(p, v, rewritePatternFn, dir, fallback[key])
+		opts := translateOptions{
+			Mode: mode,
+		}
+
+		// Try to rewrite the path relative to the directory of the configuration file where the value was defined.
+		nv, err := t.rewriteValue(ctx, p, v, dir, opts)
+		if err == nil {
+			return nv, nil
+		}
+
+		// If we failed to rewrite the path, try to rewrite it relative to the fallback directory.
+		// We only do this for jobs and pipelines because of the comment in [gatherFallbackPaths].
+		if fallback[key] != "" {
+			nv, nerr := t.rewriteValue(ctx, p, v, fallback[key], opts)
+			if nerr == nil {
+				// TODO: Emit a warning that this path should be rewritten.
+				return nv, nil
+			}
+		}
+
+		return dyn.InvalidValue, err
```

Comment:
Denormalize the previous `rewriteRelativeTo` function into the body for jobs and pipelines.

It is not relevant outside these two resources.

Reply:
Can this and below comments be incorporated into source? Seems like valuable context.

Reply:
It's included (vs the previous), here:

> 		// If we failed to rewrite the path, try to rewrite it relative to the fallback directory.
>		// We only do this for jobs and pipelines because of the comment in [gatherFallbackPaths].

From `gatherFallbackPaths` the rest of the context is clear.

I would like to deprecate it but we don't know how much usage is out there. Another good thing to track when we can capture some telemetry.